server/plugin/api.go

```diff
@@ -314,11 +314,43 @@ func (p *Plugin) completeConnectUserToGitHub(c *Context, w http.ResponseWriter,
 		return
 	}
 
-	appErr = p.API.KVDelete(stateToken)
-	if appErr != nil {
-		p.API.LogWarn("Failed to delete state token", "error", appErr.Error())
-		http.Error(w, "error deleting stored state", http.StatusBadRequest)
-		return
+	if strings.Contains(stateToken, githubTokenKey) {
+		html := `
+		<!DOCTYPE html>
+		<html>
+		<head>
+		<script>
+		window.close();
+		</script>
+		</head>
+		<body>
+		<p>Completed connecting to GitHub. Please close this window.</p>
+		</body>
+		</html>
+		`
+		if string(storedState) == "" {
+			html = `
+			<!DOCTYPE html>
+			<html>
+			<head>
+			<script>
+			window.close();
+			</script>
+			</head>
+			<body>
+			<p>Invalid Request. Please close this window.</p>
+			</body>
+			</html>
+		`
+		}
+
+		w.Header().Set("Content-Type", "text/html")
+		_, err := w.Write([]byte(html))
```

Comment:
We are now calling `w.Write` twice in this function. We should only be writing the response one time. This is true for all API requests for any plugin.

Reply:
Will follow the reference given by @hanzei  [here](https://github.com/mattermost/mattermost-plugin-github/pull/511#pullrequestreview-804104620)